modules/packages/Crypto.chpl

```diff
@@ -0,0 +1,1055 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*  A cryptographic library based on `OpenSSL <https://www.openssl.org/>`_, targeted at
+    flexible encryption purposes.
+
+    The Crypto module focuses on providing various cryptographic utilities such as
+
+    - Symmetric Ciphers
+
+      + :chpl:class:`AES`
+
+    - Asymmetric Ciphers
+
+      + :chpl:class:`RSA`
+
+    - Hashing Functions
+
+      + :chpl:class:`Hash`
+
+    - Key Derivation Functions
+
+      + :chpl:class:`KDF`
+
+    - Cryptographically-Secure Pseudo-Random Number Generators
+
+      + :chpl:class:`CryptoRandom`
+
+    - Miscellaneous Utilities.
+
+      + :chpl:class:`CryptoBuffer`
+      + :chpl:class:`RSAKey`
+      + :chpl:class:`Envelope`
+
```

Comment:
From a docs perspective, I'd still like to see a `Compiling with Crypto` section documenting the dependencies and flags to throw using paths to those dependences (openssl library and headers).

See [Compiling with BLAS](http://chapel.cray.com/docs/master/modules/packages/BLAS.html#compiling-with-blas) section for reference.

Reply:
Yep, that makes sense. Let me add that too.